### PR TITLE
[fix](profile) Should set is done when reporting profile

### PR DESCRIPTION
**Pull request by zhiqiang-hhhh**
Profile report in https://github.com/apache/doris/pull/33331 is separated into two parts: real-time & finish-time.

Profile of real-time cloud arrive FE behind the finish-time, so we need set is_done flag,  so that FE can make judge.

BTW: improve the warning log on FE.

### CODE REVIEW

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/QeProcessorImpl.java [hunk 1]
```diff
@@ -233,9 +233,13 @@ public TReportExecStatusResult reportExecStatus(TReportExecStatusParams params,
         if (params.isSetQueryProfile()) {
             if (params.isSetBackendId()) {
                 Backend backend = Env.getCurrentSystemInfo().getBackend(params.getBackendId());
+                boolean isDone = params.isSetDone() ? params.isDone() : false;
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/qe/QeProcessorImpl.java:236]: 这里是不是如果不设置，直接抛异常？
否则默认设置为true或者false，感觉逻辑上总是会有问题的。

File: fe/fe-core/src/main/java/org/apache/doris/qe/QeProcessorImpl.java [hunk 2]
```diff
@@ -233,9 +233,13 @@ public TReportExecStatusResult reportExecStatus(TReportExecStatusParams params,
         if (params.isSetQueryProfile()) {
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/qe/QeProcessorImpl.java:233]: 直接在这里check 一下is done 是否设置了，如果不设置，那么应该抛异常。这样就阻止，BE 上有没设置这个字段的情况。

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
p0 failed has nothin to do with this PR
image

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by morningman — APPROVED** [review]

**Pull request merged**